### ISSUE TITLE
AttributeTypecastBehavior uses its owner before attachment

### ISSUE THREAD

**Issue by Kolyunya** [yiisoft/yii2#12791]
Inside a behavior `$this->owner` is `NULL` at the moment of the initialization.

[Here](https://github.com/yiisoft/yii2/blob/master/framework/behaviors/AttributeTypecastBehavior.php#L197) `get_class($this->owner)` always evaluates to `yii\behaviors\AttributeTypecastBehavior` since `$this->owner` is `NULL`.

[Here](https://github.com/yiisoft/yii2/blob/master/framework/behaviors/AttributeTypecastBehavior.php#L272) a `Call to a member function getValidators() on null` error occurs for the same reason.

**Comment by dynasource** [comment]
confirmed

**Comment by klimov-paul** [comment]
I'll fix this.
In the meantime, you can bypass this issue passing `owner` via behavior config:

```php
public function behaviors()
{
     return [
         'typecast' => [
             'class' => AttributeTypecastBehavior::className(),
             'owner' => $this, // pre-fill owner
         ],
     ];
}
```

**Comment by klimov-paul** [comment]
Resolved by commit d4df72a2a4d83ce2f4ae4fef4ca252834d6a319b
